heareval/evaluation/task_evaluation.py

```diff
@@ -96,19 +175,50 @@ def task_evaluation(task_path: Path):
 
     targets = pickle.load(task_path.joinpath("test.target-labels.pkl").open("rb"))
 
-    # TODO: Check shape of predictions vs label_vocab
-
     # What other types could we receive as targets?
     assert isinstance(targets, list)
 
     # Make sure we have the same number of predictions as targets
     assert len(predictions) == len(targets)
 
+    return predictions, targets
+
+
+def get_event_based_prediction_files(task_path: Path) -> Tuple[Dict, Dict]:
+    # For event based embeddings we load JSON files of the labeled sound events
+    predictions = json.load(task_path.joinpath("test.predictions.json").open())
+    targets = json.load(task_path.joinpath("test.json").open())
+    assert isinstance(predictions, dict)
+    assert isinstance(targets, dict)
+    for filename in predictions:
+        assert filename in targets
+
+    return predictions, targets
+
+
+def task_evaluation(task_path: Path):
+
+    metadata = json.load(task_path.joinpath("task_metadata.json").open())
+    label_vocab = pd.read_csv(task_path.joinpath("labelvocabulary.csv"))
+
+    embedding_type = metadata["embedding_type"]
+
+    if "evaluation" not in metadata:
+        print(f"Task {task_path.name} has no evaluation config.")
+        return
+
+    if embedding_type == "scene":
+        predictions, targets = get_scene_based_prediction_files(task_path)
+    elif embedding_type == "event":
+        predictions, targets = get_event_based_prediction_files(task_path)
```

Comment:
This is getting a bit nasty with the event detection embeddings dir having a couple different types of labels files. Both the frame-based labels as well as the event-based labels. Perhaps in another PR we should try to clean that up a bit -- or organize the files produced in the event detection type embeddings directories.